pkg/agent/instancevalidator.go

```diff
@@ -96,22 +96,23 @@ func (j *instanceValidator) getServiceInstanceCount(attName, attValue string) in
 }
 
 func (j *instanceValidator) deleteServiceInstanceOrService(ri *apiV1.ResourceInstance, primaryKey, apiID string) {
-	msg := ""
-	var err error
-	var agentError *utilErrors.AgentError
+	// delete if it is an api service instance
+	log.Infof("API no longer exists on the dataplane, deleting the catalog item %s", ri.Title)
+	msg := "Deleted catalog item %s from Amplify Central"
 
-	defer func() {
-		// remove the api service instance from the cache for both scenarios
-		if j.isAgentPollMode {
-			// In GRPC mode delete is done on receiving delete event from service
-			agent.cacheManager.DeleteAPIServiceInstance(ri.Metadata.ID)
-		}
-	}()
+	err := agent.apicClient.DeleteAPIServiceInstance(ri.Name)
```

Comment:
May we should run the DeleteAPIs automation test to avoid any regressions.

Reply:
Will ask QA to run some tests